### ISSUE TITLE
Converged checkbox + aria-setsize and -posinset

### ISSUE THREAD

**Issue by smhigley** [microsoft/fluentui#17383]
It looks like the `react-checkbox` package supports `aria-setsize` and `aria-posinset` out of the box, even though those attributes are not supported on checkboxes.

I'd suggest removing both those attributes, and only supporting them through `inputProps`, where hopefully the author would know enough to also pass in an overriding `role` (though I can't currently think of a case where that would be a good idea).

@khmakoto does that make sense to you? I know I'm missing some of the background on this control :)

**Comment by khmakoto** [comment]
Yes, I think that makes sense, if you want to take a stab at a PR I'd be more than happy to approve it.

Just so you know, there is still work needed to be done on the `react-checkbox` package, so there might be further accessibility improvements needed (which can obviously go in a separate PR).

**Comment by smhigley** [comment]
Ah, awesome! I was just about to make a PR to add `required` (someone pointed out in a separate issue that it was missing, which is how I stumbled on this). If it's OK, I'll just combine them and add required while removing posinset and setsize :)

**Comment by smhigley** [comment]
Re-opening to track changes in v8 checkbox

**Comment by smhigley** [comment]
oops, wrong issue. Don't mind me, or my 8pm brain 🤣
